cmd/azure-config-credentials-injector/credentials_injector_test.go

```diff
@@ -0,0 +1,162 @@
+			envVars:        map[string]string{"AZURE_CLIENT_ID": "foo"},
+			expectedErrMsg: "AZURE_CLIENT_SECRET env variable should be set up",
+		},
+		{
+			name:           "input file content is not a valid json",
+			args:           []string{"--cloud-config-file-path", inputFile.Name(), "--output-file-path", outputFile.Name()},
+			envVars:        map[string]string{"AZURE_CLIENT_ID": "foo", "AZURE_CLIENT_SECRET": "bar"},
+			fileContent:    "{*&(&#@!}",
+			expectedErrMsg: "couldn't read cloud config from file: invalid character '*' looking for beginning of object key string",
+		},
+		{
+			name:           "input file content is valid json, but format is unexpected",
+			args:           []string{"--cloud-config-file-path", inputFile.Name(), "--output-file-path", outputFile.Name()},
+			envVars:        map[string]string{"AZURE_CLIENT_ID": "foo", "AZURE_CLIENT_SECRET": "bar"},
+			fileContent:    "[1]",
+			expectedErrMsg: "couldn't read cloud config from file: json: cannot unmarshal array into Go value of type map[string]interface {}",
+		},
+		{
+			name:            "all ok, file is empty json object",
+			args:            []string{"--cloud-config-file-path", inputFile.Name(), "--output-file-path", outputFile.Name()},
+			envVars:         map[string]string{"AZURE_CLIENT_ID": "foo", "AZURE_CLIENT_SECRET": "bar"},
+			fileContent:     "{}",
+			expectedContent: "{\"aadClientId\":\"foo\",\"aadClientSecret\":\"bar\"}",
+		},
+		{
+			name:            "all ok, some content in json",
+			args:            []string{"--cloud-config-file-path", inputFile.Name(), "--output-file-path", outputFile.Name()},
+			envVars:         map[string]string{"AZURE_CLIENT_ID": "foo", "AZURE_CLIENT_SECRET": "bar"},
+			fileContent:     "{\"bar\": \"baz\"}",
+			expectedContent: "{\"aadClientId\":\"foo\",\"aadClientSecret\":\"bar\",\"bar\":\"baz\"}",
+		},
+		{
+			name:            "all ok, client_id and client_secret overrides",
+			args:            []string{"--cloud-config-file-path", inputFile.Name(), "--output-file-path", outputFile.Name()},
+			envVars:         map[string]string{"AZURE_CLIENT_ID": "foo", "AZURE_CLIENT_SECRET": "bar"},
+			fileContent:     "{\"aadClientSecret\":\"fizz\",\"aadClientId\":\"baz\"}",
+			expectedContent: "{\"aadClientId\":\"foo\",\"aadClientSecret\":\"bar\"}",
+		},
+		{
+			name:           "output file write error",
+			args:           []string{"--cloud-config-file-path", inputFile.Name(), "--output-file-path", "/tmp"},
+			envVars:        map[string]string{"AZURE_CLIENT_ID": "foo", "AZURE_CLIENT_SECRET": "bar"},
+			fileContent:    "{}",
+			expectedErrMsg: "couldn't write prepared cloud config to file: open /tmp: is a directory",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			for envVarName, envVarValue := range tc.envVars {
+				if err := os.Setenv(envVarName, envVarValue); err != nil {
+					t.Fatalf("Can not setup environment variable %s: %v", envVarName, err)
+				}
```

Comment:
Could use require here too right?
```suggestion
				err := os.Setenv(envVarName, envVarValue)
				require.NoError(t, err, "Can not setup environment variable %s: %v", envVarName, err)
				
```

Reply:
Why was this marked resolved?